sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/implementation/IngestionUsingDataCollectionRulesClientBuilder.java

```diff
@@ -250,21 +255,19 @@ private HttpPipeline createHttpPipeline() {
         if (headers.getSize() > 0) {
             policies.add(new AddHeadersPolicy(headers));
         }
-        policies.addAll(
-                this.pipelinePolicies.stream()
-                        .filter(p -> p.getPipelinePosition() == HttpPipelinePosition.PER_CALL)
-                        .collect(Collectors.toList()));
+        this.pipelinePolicies.stream()
+                .filter(p -> p.getPipelinePosition() == HttpPipelinePosition.PER_CALL)
+                .forEach(p -> policies.add(p));
         HttpPolicyProviders.addBeforeRetryPolicies(policies);
         policies.add(ClientBuilderUtil.validateAndGetRetryPolicy(retryPolicy, retryOptions, new RetryPolicy()));
         policies.add(new AddDatePolicy());
         policies.add(new CookiePolicy());
         if (tokenCredential != null) {
             policies.add(new BearerTokenAuthenticationPolicy(tokenCredential, DEFAULT_SCOPES));
```

Comment:
Should we only add the scope that is needed for that particular cloud env rather than adding all of them?

Reply:
@srnagar Since this builder is autogenerated I'd refrain from doing that since that would require manually editing the code each time after generation or writing a code customization to insert the logic which is unnecessary technical debt in my opinion.

Reply:
@srnagar Unless we also decide to replace the use of this autogenerated builder with a handwritten one, I don't think we should change how it's been generated. What do you think about that option?

Reply:
@srnagar do you suggest I adjust the public API and write a code customization to replace the logic?

Reply:
@jairmyree yes, let's add code customization and limit the scope added to only the specific cloud environment used by the client.